quantum/rgb_matrix.c

```diff
@@ -427,14 +448,26 @@ void rgb_matrix_init(void) {
     rgb_matrix_config.raw = eeconfig_read_rgb_matrix();
   }
   eeconfig_debug_rgb_matrix(); // display current eeprom values
+
+#ifdef RGB_MATRIX_MULTIPLE_EFFECTS
+  rgb_effect_params.flags = rgb_matrix_flags_multi[0];
+  for(uint8_t i = 0; i < RGB_MATRIX_MULTIPLE_COUNT; i++) {
+    rgb_matrix_config_multi[i] = rgb_matrix_config;
+    rgb_effect_params_multi[i] = rgb_effect_params;
+    rgb_effect_params_multi[i].flags = rgb_matrix_flags_multi[i + 1];
+    rgb_last_effect_multi[i] = UINT8_MAX;
+  }
+#endif
 }
 
 void rgb_matrix_set_suspend_state(bool state) {
   g_suspend_state = state;
+  if (!g_suspend_state)
+    rgb_task_state = STARTING;
 }
 
 void rgb_matrix_toggle(void) {
-  rgb_matrix_config.enable ^= 1;
+  g_rgb_matrix_enable = !g_rgb_matrix_enable;
```

Comment:
If this is no longer part of the structure, there is no reason to write to eeprom